tests/contrib/django/test_django_snapshots.py

```diff
@@ -98,3 +185,35 @@ def test_psycopg_query_default(client, psycopg2_patched):
         rows = cur.fetchall()
         assert len(rows) == 1, rows
         assert rows[0][0] == "one"
+
+
+@pytest.mark.skipif(django.VERSION < (3, 0, 0), reason="ASGI not supported in django<3")
+@snapshot(
+    async_mode=False,
+    variants={
+        "30": (3, 0, 0) <= django.VERSION < (3, 1, 0),
+        "31": (3, 1, 0) <= django.VERSION < (3, 2, 0),
+        "": django.VERSION >= (3, 2, 0),
+    },
+)
+def test_asgi_200(daphne_client):
+    resp = daphne_client.get("/")
+    assert resp.status_code == 200
+    assert resp.content == b"Hello, test app."
+    time.sleep(0.1)
```

Comment:
Introduce an endpoint to force a flush of the traces to reduce this flakiness. Thanks @P403n1x87!

Reply:
Interesting idea! We'd want to disable tracing for such an endpoint so we don't end up getting it's trace in the snapshot as well.

Reply:
Another option would be to have the client fixture be per test such that flushing occurs during shutdown.

Reply:
> We'd want to disable tracing for such an endpoint so we don't end up getting it's trace in the snapshot as well.

Ah yes... good point!

> Another option would be to have the client fixture be per test such that flushing occurs during shutdown.

yes - this would be more ideal IMO but there is a non-trivial cost of starting up the webserver unfortunately... although we don't run a lot of test scenarios

Reply:
glad to say that all the `time.sleep()`s have been purged! Instead,

- the django instance is started per process 
- traces are sent with a test token directly thanks to #2689 
- polling spans are filtered by using http propagation headers with a special trace id
- `TraceFilter` filters traces with the special trace id 
- traces are force-flushed with a call to a special endpoint